internal/sign.go

```diff
@@ -0,0 +1,39 @@
+package internal
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/invopop/gobl"
+	"github.com/labstack/echo/v4"
+)
+
+// Sign parses a GOBL document into an envelope, performs calculations,
+// validates it, and finally signs its headers. The parsed envelope *must* be a
+// draft, or else an error is returned.
+func Sign(ctx context.Context, opts *BuildOptions) (*gobl.Envelope, error) {
+	// TODO: `BuildOptions` should probably be renamed to `ParseOptions`,
```

Comment:
Is this `TODO` valid, and if so, do we want to include it in this PR?

Reply:
I think this is related to your previous comment and renaming "BuildOptions"... they don't really reflect what is happening.